stdlib/filename.mli

```diff
@@ -49,13 +49,42 @@ val chop_suffix : string -> string -> string
    the filename [name]. The behavior is undefined if [name] does not
    end with the suffix [suff]. *)
 
+val extension : string -> string
+(** [extension name] is the shortest suffix [ext] of [name0] where:
+
+    - [name0] is the longest suffix of [name] that does not
+      contain a directory separator;
+    - [ext] starts with a period;
+    - [ext] is preceded by at least one non-period character
+      in [name0].
+
+    If such a suffix does not exist, [extension name] is the empty
+    string.
+
+    @since 4.04
+*)
+
+
+
 val chop_extension : string -> string
-(** Return the given file name without its extension. The extension
-   is the shortest suffix starting with a period and not including
-   a directory separator, [.xyz] for instance.
+(** Return the given file name without its extension, as defined
+    in {!Filename.extension}.
+
+    Raise [Invalid_argument] if the given name has an empty extension. *)
+
+val chop_extension_if_any : string -> string
```

Comment:
I'm ok with both proposals (deprecating and renaming)., but I'm wondering about the effects about changing the semantics of `chop_extension`.  Considering that `Invalid_argument` is indeed for programming errors, one could say that returning something different for that case should not break any correct code.  Well, there are occurrences, even in the core distrib, of code that captures explicitly that `Invalid_argument`, but this is always to return the original filename (i.e. the new suggested behavior).  Opinions about changing the semantics of `chop_extension` instead of introducing a new function?